T3/Gui/Graph/Interaction/SymbolBrowser.cs

```diff
@@ -287,12 +293,129 @@ private void DrawMatchesList()
 
             ImGui.EndChildFrame();
 
-            DrawDescriptionPanel(itemForHelp);
-
             ImGui.PopStyleVar(2);
         }
 
-        private static void DrawDescriptionPanel(SymbolUi itemForHelp)
+        private SymbolUi DetermineItemForHelp(SymbolUi symbolUi, bool isSelected, bool isHovered)
+        {
+            if (isHovered)
+            {
+                _lastHoveredSymbolUi = symbolUi;
+                return symbolUi;
+            }
+
+            if (symbolUi != _lastHoveredSymbolUi)
+                return isSelected ? symbolUi : null;
+            
+            if (UserSettings.Config.SymbolBrowserDescriptionTimeout)
+            {
+                ExpireLastHoveredSymbolUi(symbolUi);
+            }
+
+            return symbolUi;
+
+        }
+
+        /// <summary>
+        /// A method to give some time before expiring a hovered symbol.
+        /// This is intended to allow the cursor to cross over the scroll bar
+        /// of the symbol browser so examples can be interacted with
+        /// </summary>
+        /// <param name="hoveredSymbolUi"></param>
+        async void ExpireLastHoveredSymbolUi(SymbolUi hoveredSymbolUi)
```

Comment:
Wow. It took me a while to understand what this method is doing. 
Do we really need such a complex solution?

Reply:
I think you have to feel more comfortable with immediate user interface: The method is called every frame. So it would be MUCH easier to just store the start time and the check on every frame (I.e. on every redraw) if the time has exceeded. i'm pretty sure there is no need for async here.

Reply:
yes it's very complex, I was wrestling with this for a while and I'm also not a huge fan of it. In retrospect this can very much be done in-line or non-async, but I liked the way the async method kept the timing logic and variables in one place. if this was that difficult for you of all people, i think it's definitely worth doing the other way

I can also remove the expiration-timer feature altogether - I implemented this in the case that you didnt want the symbol browser to keep the description open when the symbol browser was up